File: 2022/day2/part1.js
const fs = require("fs");
var score = 0;

fs.readFile("input.txt", 'utf8', (err, data) => {
	data.split("\n").forEach(line => {
		var oppoMove = line.split(" ")[0];
		var myMove = line.split(" ")[1];

		switch (myMove){
			case "X":
				score += 1;
				break;
			case "Y": 
				score += 2;
				break;
			case "Z":
				score += 3;
				break;
		}

		switch (line){
			case "A X":
			case "B Y":
			case "C Z":
				score += 3;
				break;
			case "A Y":
			case "B Z":
			case "C X":
				score += 6;
				break;
			case "A Z":
			case "B X":
			case "C Y":
				score += 0;
				break;
		}
	})

	console.log(score);
})

File: 2023/day4/part1.py
file = open('input.txt', 'r')
lines = file.readlines()

total = 0

for line in lines:
	line = line[line.find(":")+2:len(line)]

	winners = line.split("|")[0].split(" ")
	winners = [i.strip() for i in winners if i]

	numbers = line.split("|")[1].split(" ")
	numbers = [i.strip() for i in numbers if i]

	score = 0
	matches = set(winners) & set(numbers)
	#print(str(len(matches)))

	for x in range(len(matches)):
		if x == 0:
			score = score + 1
		else:
			score = score * 2

	total = total + score

print(total)